flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -225,10 +228,14 @@ Map<JobVertexID, ScalingSummary> computeScalingSummary(
                             } else {
                                 var currentParallelism =
                                         (int) metrics.get(ScalingMetric.PARALLELISM).getCurrent();
+
+                                final boolean hasKeyBy =
+                                        jobTopology.get(v).getInputs().containsValue(HASH);
```

Comment:
I assume this also means that sources are also not adjusted which can lead to partition imbalance for Kafka topics. We should either keep the adjustments for sources or remove the whole partition number querying/max parallelism logic that we have during collection. 

Overall I think for sources this makes sense and if we want to remove it that should be a separate ticket.

Reply:
This is related to https://issues.apache.org/jira/browse/FLINK-32119

Reply:
Thanks a lot for pointing it out. I forgot it before.

I have updated. The source can adjust parallelism for now. Also, I updated parameter names and the comment for `JobVertexScaler#scale` method, for exampe: changing the numKeyGroups to maxParallelism, and changing hasKeyBy to adjustByMaxParallelism.

Because the numKeyGroups and hasKeyBy aren't exactly for source vertex.